cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/FindPluginHelper.java

```diff
@@ -69,6 +70,42 @@ public static Plugin findPlugin(ArtifactRepository artifactRepository,
     } catch (IOException e) {
       throw Throwables.propagate(e);
     }
+    return getPlugin(pluginEntry, properties, pluginType, pluginName, pluginInstantiator);
```

Comment:
this entire method should call the new method and just convert the single id to a range.